python/lance/_lib.pyx

```diff
@@ -220,6 +279,28 @@ cdef class FileSystemDataset(Dataset):
         c_version = GetResultValue(self.lance_dataset.latest_version())
         return _dataset_version_to_json(c_version)
 
+    def append_column(self, field: Field, func: Callable[[pyarrow.Table], pyarrow.Array]) -> FileSystemDataset:
```

Comment:
User needs to know that func input is just a chunk of data. Eg they can't do something like normalization in this function unless they explicitly compute the mean / std globally.

Reply:
k, will do.

Reply:
This is actually a good argument of whether func should take in a RecordBatch instead of Table